sw/device/tests/sram_ctrl_smoketest.c

```diff
@@ -0,0 +1,88 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+// define a number of reads and writes to perform,
+// for our purposes a small number will be sufficient.
+static const uint32_t kTestDataSizeWords = 16;
+
+static const uint32_t kStatusRegMask = kDifSramCtrlStatusBusIntegErr |
+                                       kDifSramCtrlStatusInitErr |
+                                       kDifSramCtrlStatusEscalated;
+
+static const uint32_t kRandomData[kTestDataSizeWords] = {
+    0x6b4abfae, 0x63bdb6e7, 0x87f99b1a, 0xa214dffe, 0xb12291f9, 0xd0cd1abe,
+    0x5c95e716, 0xe887aab1, 0x307f6ef9, 0x6f5c0464, 0x5882279d, 0x44c19574,
+    0x1bd20079, 0xf8250ead, 0x4bf362a4, 0xad41437d};
+
+// Buffer to allow the compiler to allocate a safe area in Main SRAM where
+// we can do the write/read test without the risk of clobbering data
+// used by the program.
+OT_ATTR_SECTION(".data")
+static volatile const uint32_t sram_main_buffer[kTestDataSizeWords];
```

Comment:
I guess technically it should not be `const`, as you do write into this buffer (even though not doing so directly).

Reply:
No problem, I have dropped the const.
Many thanks, indeed you have been patient and very helpful :-)

Reply:
No problem, likewise, thank you for your patience as well :)